cmake/thirdparty/SetupHIP.cmake

```diff
@@ -15,24 +15,76 @@ find_package(HIP REQUIRED)
 message(STATUS "HIP version:      ${HIP_VERSION_STRING}")
 message(STATUS "HIP platform:     ${HIP_PLATFORM}")
 
-if("${HIP_PLATFORM}" STREQUAL "hcc" OR "${HIP_PLATFORM}" STREQUAL "amd")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_HCC__")
-elseif("${HIP_PLATFORM}" STREQUAL "nvcc")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_NVCC__")
+set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include")
+if(${HIP_PLATFORM} STREQUAL "hcc")
+	set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libhip_hcc.so")
+elseif(${HIP_PLATFORM} STREQUAL "clang" OR ${HIP_PLATFORM} STREQUAL "amd")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__;-D__HIP_ROCclr__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libamdhip64.so")
```

Comment:
Is this the only runtime library required, and is it always shared?

Reply:
`libamdhip64` is the only direct dependency for the current (since ROCM 3.5) HIP runtime. In ROCM binary distributions all the dependencies of `libamdhip64` are runpathe'ed, and in spack ROCM builds they're RPATH'ed, so it should be sufficient to just link to `libamdhip64`.

As far as I'm aware, AMD only provides it as a shared object. I've not seen nor been able to find a static HIP runtime library.

Reply:
Can you change this to a `find_library` call?  Not only am I worried about future proofing against a static library but it would also handle the `lib64` case naturally...  This should be sufficent:

```
# Find libraries
find_library(HIP_RUNTIME_LIBRARIES NAMES amdhip64 libamdhip64
                   PATHS ${HIP_ROOT_DIR}/lib
                   NO_DEFAULT_PATH
                   NO_CMAKE_ENVIRONMENT_PATH
                   NO_CMAKE_PATH
                   NO_SYSTEM_ENVIRONMENT_PATH
                   NO_CMAKE_SYSTEM_PATH)
```

Reply:
If the `find_library` approach ends up not being feasible, perhaps it would also be preferable to use [shared prefix](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SHARED_LIBRARY_PREFIX.html) and [shared suffix](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SHARED_LIBRARY_SUFFIX.html) variables for generality.

Reply:
`find_library` worked fine here. I've adapted everything and tested what I can. I don't have any of the old `hcc` based ROCm installs around (needs to be <ROCM 3.5), but I don't see any reason why that path shouldn't also work.